analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -155,6 +156,11 @@ public override void VisitAttribute(AttributeSyntax node)
                 {
                     TypesUsedWithReflection.Add(typeSymbol);
                 }
+                else if (symbol.ContainingType.GetEffectiveAccessibility() == Accessibility.Private
```

Comment:
The method uses `symbol.ContainingType` a lot. You may want to change the above for better readability
```cs
if (semanticModel.GetSymbolInfo(node).Symbol is IMethodSymbol { MethodKind: Constructor, ContainingType: ITypeSymbol attribute } attributeConstructor)
```